File: 0828/828.go
func countUniqueCharactersOfAllSubs(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := countUniqueCharactersOfAllSubs(root.Left)
	r := countUniqueCharactersOfAllSubs(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0828/828.cpp
class Solution {
public:
    int countUniqueCharactersOfAllSubs(TreeNode* root) {
        if(!root) return 0;
        return 1+max(countUniqueCharactersOfAllSubs(root->left),countUniqueCharactersOfAllSubs(root->right));
    }
};